packages/x-date-pickers-pro/src/models/fields.ts

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import { SlotComponentProps } from '@mui/base/utils';
+import { BaseFieldProps, UseFieldResponse } from '@mui/x-date-pickers/internals';
+import {
+  BaseSingleInputPickersTextFieldProps,
+  FieldRef,
+  FieldSection,
+  PickerValidDate,
+} from '@mui/x-date-pickers/models';
+import { UseClearableFieldResponse } from '@mui/x-date-pickers/hooks';
+import { SxProps } from '@mui/material/styles';
+
+export interface RangeFieldSection extends FieldSection {
+  dateName: 'start' | 'end';
+}
+
+export type FieldType = 'single-input' | 'multi-input';
+
+/**
+ * Props the `textField` slot of the multi input field can receive when used inside a picker.
+ */
+export interface MultiInputFieldSlotTextFieldProps {
+  label?: React.ReactNode;
+  id?: string;
+  disabled?: boolean;
+  readOnly?: boolean;
+  onKeyDown?: React.KeyboardEventHandler;
+  onClick?: React.MouseEventHandler;
+  onFocus?: React.FocusEventHandler;
+  focused?: boolean;
+  InputProps?: {
+    ref?: React.Ref<any>;
+    endAdornment?: React.ReactNode;
+    startAdornment?: React.ReactNode;
+  };
+}
+
+/**
+ * Props the `root` slot of the multi input field can receive when used inside a picker.
+ */
+export interface MultiInputFieldSlotRootProps {
+  onBlur?: React.FocusEventHandler;
+}
+
+export interface MultiInputFieldRefs {
+  unstableStartFieldRef?: React.Ref<FieldRef<RangeFieldSection>>;
+  unstableEndFieldRef?: React.Ref<FieldRef<RangeFieldSection>>;
+}
+
+/**
+ * Props the multi input field can receive when used inside a picker.
+ * Only contains what the MUI components are passing to the field,
+ * not what users can pass using the `props.slotProps.field`.
+ */
+export interface BaseMultiInputFieldProps<
+  TValue,
+  TDate extends PickerValidDate,
+  TSection extends FieldSection,
+  TEnableAccessibleFieldDOMStructure extends boolean,
+  TError,
+> extends Omit<
+    BaseFieldProps<TValue, TDate, TSection, TEnableAccessibleFieldDOMStructure, TError>,
+    'unstableFieldRef'
+  > {
+  sx?: SxProps<any>;
```

Comment:
Nitpick: Shouldn't this be typed with `Theme` as all the other instances? 🤔 
```suggestion
  sx?: SxProps<Theme>;
```

Reply:
If we do, the example with Joy are complaining because the theme of Joy is not the same :+1:

Reply:
Hmm, that is an interesting problem. 🙈 
Maybe @siriwatknp has any idea for an intermediate solution? 🤔 
It seems that the `SxProps` from `Joy` is correctly typed with `Joy` `Theme`.
So, technically, this can be an annoying shortcoming. 🤷

Reply:
To summarize for Jun

We have the field component which receives some props and passes them to its textfield.
We have the picker component which renders the fields with some props, including `sx` (when doing `<DatePicker sx={{}} />`, the `sx` prop is forwarded to the field)
We have an interface that describes what the field receives from the picker, to help people building custom field without forgetting some callbacks / some ref / etc... this interface includes the `sx` prop since the pickers passes it to the field.

For now, we have a single `DatePicker` component which can be customized through a slot to accept a field built with the Joy UI textfield.
So the interface can use `sx?: SxProps<Theme>` because `Theme` is the material theme and not compatible when the field is built for Joy.

The current customization is clearly not optimal and for advanced customization like a Joy UI textfield, it would be better to expose a `DatePicker` built with Joy UI.